cpp/include/cudf/ast/detail/expression_evaluator.cuh

```diff
@@ -278,7 +278,7 @@ struct expression_evaluator {
     detail::device_data_reference const& input_reference,
     IntermediateDataType<has_nulls>* thread_intermediate_storage,
     cudf::size_type left_row_index,
-    thrust::optional<cudf::size_type> right_row_index = {}) const
+    cuda::std::optional<cudf::size_type> right_row_index = {}) const
```

Comment:
@bdice Do you think we could remove the `optional` here? There is a comment below that says in some cases `right_row_index` is ignored. I think just making this default to `{}` is enough to make this parameter _optional_ without making it a formal `cuda::std::optional`.

Reply:
Exploring this in https://github.com/rapidsai/cudf/pull/16604.